Upgrade otel/build-protobuf image 0.2.1->0.11.0

MrAlias opened the pull request:
Fix #45 

- Upgrades `github.com/grpc-ecosystem/grpc-gateway` v1.16.0 -> v2.7.0 (last version that supports Go 1.14 and resolves vulnerability)
- Switch from `github.com/golang/protobuf` to `google.golang.org/protobuf` entirely.

MrAlias commented:
Looking at options to address Go 1.17 min requirement. Might be worth following suit for the repo.

Aneurysm9 commented:
Can we use `github.com/grpc-ecosystem/grpc-gateway@v2.7.0`?  That seems to be the most recent version that doesn't explicitly require 1.17.

MrAlias commented:
> Can we use `github.com/grpc-ecosystem/grpc-gateway@v2.7.0`? That seems to be the most recent version that doesn't explicitly require 1.17.

It doesn't look like there is a release that supports that version. [`v0.8.0`](https://hub.docker.com/layers/build-protobuf/otel/build-protobuf/0.8.0/images/sha256-df3d1d27333412da7dbaa8be0d483963b4865b4975bf31c8f003866fc04e278e?context=explore) brings in these dependencies:

```sh
go: creating new go.mod: module go.opentelemetry.io/proto/otlp
go: to add module requirements and sums:
	go mod tidy
go: module github.com/golang/protobuf is deprecated: Use the "google.golang.org/protobuf" module instead.
go: added github.com/golang/protobuf v1.5.2
go: added github.com/grpc-ecosystem/grpc-gateway v1.16.0
go: added google.golang.org/grpc v1.45.0
go: added google.golang.org/protobuf v1.28.0
```
and [`v0.9.0`](https://hub.docker.com/layers/build-protobuf/otel/build-protobuf/0.9.0/images/sha256-e561f2344b90314714aa27efe684de277a17b83867c6ae72b2ad0e7f9e0707d2?context=explore):

```sh
go: creating new go.mod: module go.opentelemetry.io/proto/otlp
go: to add module requirements and sums:
	go mod tidy
go: added github.com/grpc-ecosystem/grpc-gateway/v2 v2.10.0
go: added google.golang.org/grpc v1.45.0
go: added google.golang.org/protobuf v1.28.0
```

MrAlias commented:
Given there is a security issue this is resolving, it seems to motivate moving forward to Go 1.17 early.

MrAlias commented:
@Aneurysm9 PTAL. I think I've figured out a way to upgrade reliably to where we want `github.com/grpc-ecosystem/grpc-gateway` and keep backwards compatible support.
